build-common/src/org/jetbrains/kotlin/incremental/storage/externalizers.kt

```diff
@@ -384,11 +323,13 @@ abstract class GenericCollectionExternalizer<T, C : Collection<T>>(
     }
 }
 
+object StringCollectionExternalizer : CollectionExternalizer<String, Collection<String>>(StringExternalizer, { size -> ArrayList(size) })
```

Comment:
I'm a bit worried by that change. Previously it was backed by `HashSet`

Reply:
A **Collection**Externalizer must be able to preserve duplicates in the collection, so I think using a List is required for correctness. Using a `HashSet` before was probably a mistake.

Reply:
Yep, I agree that in general it should be as you did. Since you're going to perform more work on that as you say below, then I'm fine with that

Reply:
Great, thank you for your review! :)